### ISSUE TITLE
Duplicate SSA Error Message Returning in Go SDK

### ISSUE THREAD

**Issue by zbuchheit** [pulumi/pulumi-kubernetes#2643]
### What happened?

While attempting to do a DaemonSetPatch in Go, and the annotation `pulumi.com/patchForce` isn't set, I receive a duplicate error message when changing the image that there are multiple conflicts. The error I receive is as follows below.

```
    error: resource kube-system/kube-proxy was not successfully created by the Kubernetes API server : Server-Side Apply field conflict detected. see https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/managing-resources-with-server-side-apply/#handle-field-conflicts-on-existing-resources for troubleshooting help
    : Apply failed with 2 conflicts: conflicts with "kubectl" using apps/v1:
    - .spec.template.spec.containers[name="kube-proxy"].image
    - .spec.template.spec.containers[name="kube-proxy"].image
```

### Example

```
		_, err = v1.NewDaemonSetPatch(ctx, "k8s-daemonset-patch", &v1.DaemonSetPatchArgs{
					Metadata: metav1.ObjectMetaPatchArgs{
						Name:      pulumi.String("kube-proxy"),
						Namespace: pulumi.String("kube-system"),
						Annotations: pulumi.StringMap{
							// "pulumi.com/patchForce":        pulumi.String("true"),
						},
					},
					Spec: v1.DaemonSetSpecPatchArgs{
						Template: corev1.PodTemplateSpecPatchArgs{
							Spec: corev1.PodSpecPatchArgs{
								Containers: corev1.ContainerPatchArray{
									corev1.ContainerPatchArgs{
										Name:  pulumi.String("kube-proxy"),
										Image: pulumi.String("602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/kube-proxy:v1.24.17-minimal-eksbuild.2"), //update from 1.24.7
									},
								},
							},
						},
					},
				}, pulumi.Provider(kubeProvider), 
				pulumi.RetainOnDelete(true), 
			)
		if err != nil {
			return err
		}
```

### Output of `pulumi about`

```
CLI          
Version      3.90.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.35.0
docker      3.2.0
eks         1.0.3
go          unknown
kubernetes  4.1.1

Host     
OS       darwin
Version  14.0
Arch     arm64

This project is written in go: executable='/opt/homebrew/bin/go' version='go version go1.21.0 darwin/arm64'

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/zbuchheit-pulumi-corp
User           zbuchheit-pulumi-corp
Organizations  zbuchheit-pulumi-corp
Token type     personal

Dependencies:
NAME                                        VERSION
github.com/pulumi/pulumi-eks/sdk            1.0.3
github.com/pulumi/pulumi-kubernetes/sdk/v4  4.1.1
github.com/pulumi/pulumi/sdk/v3             3.90.0
```

### Additional context

⚠️ This error was not present when using Typescript sdk and only appeared when I switched to Go. It displayed 1 conflict as expected when using Typescript

`Pulumi preview` works successfully, but errors on `Pulumi up`. If there are other conflicts it will also show duplicates of those as well, but for this example Image was the simplest example to provide.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by rquitales** [comment]
Thanks for filing this bug. I can confirm this affects our Go SDK, and have a potential fix in mind for this.

**Comment by rquitales** [comment]
Upon further investigation, it appears that the issue stems from an upstream error in the Kubernetes API server rather than a flaw in our error handling. My earlier assumption about a specific code branch causing the problem was mistaken. The duplicate error we're encountering is directly from the Kubernetes cluster's API server itself, and there is no formatting on our end contributing to the duplication.

I've successfully reproduced this issue with other daemonset objects across various Kubernetes environments (GKE, EKS, Kind) and different versions. Additionally, I was able to replicate the duplicated error message using standard Kubernetes yaml manifests, confirming that the root cause lies beyond the control of Pulumi.

I've identified an upstream issue that precisely describes this behavior: https://github.com/kubernetes/kubernetes/issues/119141. It seems that the fix for this problem will be available in v1.29 of Kubernetes. Currently, there is no indication of this fix being backported to earlier versions. Considering this, I suggest closing this issue as it is beyond the scope of Pulumi's control.
